src/commands/deploy/validateRemoteBuild.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { SiteClient } from 'vscode-azureappservice';
+import { IActionContext } from 'vscode-azureextensionui';
+import { deploySubpathSetting, packTaskName, preDeployTaskSetting, ProjectLanguage } from '../../constants';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../../vsCodeConfig/settings';
+import { tryGetFunctionProjectRoot } from '../createNewProject/verifyIsProject';
+import { ensureGitIgnoreContents } from '../initProjectForVSCode/InitVSCodeStep/PythonInitVSCodeStep';
+
+export async function validateRemoteBuild(context: IActionContext, client: SiteClient, workspacePath: string, language: ProjectLanguage): Promise<void> {
+    if (language === ProjectLanguage.Python && !client.kuduUrl) {
+        const remoteBuildKey: string = 'scmDoBuildDuringDeployment';
+
+        const remoteBuild: boolean | undefined = getWorkspaceSetting(remoteBuildKey, workspacePath);
+        if (remoteBuild) {
+            const message: string = localize('remoteBuildNotSupported', 'The selected Function App doesn\'t support your project\'s configuration. Deploy to a newer Function App or downgrade your config.');
```

Comment:
How does it know that the function app doesn't support remote build?  Does it only work with kudu?

Reply:
Yes it only works if it has kudu (I mean it's probably oryx, but we treat those the same [in SiteClient](https://github.com/microsoft/vscode-azuretools/blob/master/appservice/src/SiteClient.ts#L77))